python-frontend/src/test/java/org/sonar/python/tree/CallExpressionImplTest.java

```diff
@@ -202,4 +207,20 @@ public void inner_class_constructor_declared_type() {
     assertThat(type).isInstanceOf(DeclaredType.class);
     assertThat(typeName(type)).isEqualTo("B");
   }
+
+  @Test
+  public void test_generic_collections() {
+    assertThat(lastExpressionInFunction("list[int]()").type()).isEqualTo(LIST);
+    assertThat(lastExpressionInFunction("tuple[int]()").type()).isEqualTo(TUPLE);
+    assertThat(lastExpressionInFunction("dict[str, int]()").type()).isEqualTo(DICT);
+    assertThat(lastExpressionInFunction("set[int]()").type()).isEqualTo(SET);
+    assertThat(lastExpressionInFunction("frozenset[int]()").type().canOnlyBe("frozenset")).isTrue();
+    assertThat(lastExpressionInFunction("from collections import deque; deque[int]()").type().canOnlyBe("collections.deque")).isTrue();
+    assertThat(lastExpressionInFunction("from collections import defaultdict; defaultdict[str, int]()").type().canOnlyBe("collections.defaultdict")).isTrue();
+    assertThat(lastExpressionInFunction("from collections import OrderedDict; OrderedDict[str, int]()").type().canOnlyBe("collections.OrderedDict")).isTrue();
+    assertThat(lastExpressionInFunction("from collections import Counter; Counter[int]()").type().canOnlyBe("collections.Counter")).isTrue();
+    assertThat(lastExpressionInFunction("from collections import ChainMap; ChainMap[str, int]()").type().canOnlyBe("collections.ChainMap")).isTrue();
+    assertThat(lastExpressionInFunction("from foo import bar; bar[str]()").type().canOnlyBe("bar.foo")).isFalse();
```

Comment:
Shouldn't it be `.canOnlyBe("foo.bar")`?
In fact, it may be better to assert that it's equal to `ANY`.